test_unit.py

```diff
@@ -1209,6 +1209,35 @@ def test_HTLC(self):
         self.assertEqual(p, base64.b64decode(golden))
         self.assertEqual(s.get_address(), golden_addr)
 
+    def test_dynamic_fee(self):
+        addr1 = "726KBOYUJJNE5J5UHCSGQGWIBZWKCBN4WYD7YVSTEXEVNFPWUIJ7TAEOPM"
+        addr2 = "42NJMHTPFVPXVSDGA6JGKUV6TARV5UZTMPFIREMLXHETRKIVW34QFSDFRE"
+        s = template.DynamicFee(addr1, 5000, 12345, 12346, addr2)
+        s.lease_value = base64.b64decode("f4OxZX/x/FO5LcGBSKHWXfwtSx+j1ncoSt3SABJtkGk=")
+
+        golden_addr = ("GCI4WWDIWUFATVPOQ372OZYG52EUL" +
+                       "PUZKI7Y34MXK3ZJKIBZXHD2H5C5TI")
+
+        golden = ("ASAFAgGIJ7lgumAmAyD+vKC7FEpaTqe0OKRoGsgObKEFvLYH/FZTJclW" +
+                  "lfaiEyDmmpYeby1feshmB5JlUr6YI17TM2PKiJGLuck4qRW2+SB/g7Fl" +
+                  "f/H8U7ktwYFIodZd/C1LH6PWdyhK3dIAEm2QaTIEIhIzABAjEhAzAAcx" +
+                  "ABIQMwAIMQESEDEWIxIQMRAjEhAxBygSEDEJKRIQMQgkEhAxAiUSEDEE" +
+                  "IQQSEDEGKhIQ")
```

Comment:
The golden I've created doesn't match this one, I recreated the program with `algotmpl` and it doesn't match either:
```
# Dynamic Fee
~$ algotmpl -d ~/${GOPATH}/src/github.com/algorand/go-algorand/tools/teal/templates/ \
    dynamic-fee \
    --amt 5000 \
    --cls 42NJMHTPFVPXVSDGA6JGKUV6TARV5UZTMPFIREMLXHETRKIVW34QFSDFRE \
    --to 726KBOYUJJNE5J5UHCSGQGWIBZWKCBN4WYD7YVSTEXEVNFPWUIJ7TAEOPM \
    --fv 12345 \
    --lv 123456 \
    --lease "f4OxZX/x/FO5LcGBSKHWXfwtSx+j1ncoSt3SABJtkGk=" \
    > dynamic-fee-golden.teal
~$ goal clerk compile dynamic-fee-golden.teal -o dynamic-fee-golden.bin
dynamic-fee-golden.teal: JKTF262LODA6NJLD64YR3QJ34SSYQT2OT6RWKY57HKG2KZMGGLC4CYBA2E
~$ base64 dynamic-fee-golden.bin
ASAFAgGIJ7lgwMQHJgMg/ryguxRKWk6ntDikaBrIDmyhBby2B/xWUyXJVpX2ohMg5pqWHm8tX3rI
ZgeSZVK+mCNe0zNjyoiRi7nJOKkVtvkgf4OxZX/x/FO5LcGBSKHWXfwtSx+j1ncoSt3SABJtkGky
BCISMwAQIxIQMwAHMQASEDMACDEBEhAxFiMSEDEQIxIQMQcoEhAxCSkSEDEIJBIQMQIlEhAxBCEE
EhAxBioSEA==
```

Reply:
False alarm! I had lastValid wrong in the example above.